x-pack/plugins/apm/public/components/app/TransactionDetails/Transaction/WaterfallContainer/Waterfall/waterfall_helpers/waterfall_helpers.ts

```diff
@@ -231,43 +236,44 @@ function createGetTransactionById(itemsById: IWaterfallIndex) {
 }
 
 export function getWaterfall(
```

Comment:
This who method seems like it could use a bit of cleanup. There are several different data structures that might benefit from a class-based approach. If you have any idea, feel free to try it out, else we can just leave it for another day.

Reply:
i started down the path of refactoring portions of these functions and data structures, but there are deep waterfall and flyout components which currently depend on the existing pattern, and it started to take more time that i think is appropriate for this PR. I think we should tackle this as a tech debt issue to simplify this data flow.